### PR TITLE
HPCC4J-457 Correctly sets isHTTPs=false  inConnection.setProtocol

### PR DESCRIPTION
**Pull request by rpastrana**
Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [x] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

File: wsclient/src/test/java/org/hpccsystems/ws/client/utils/ConnectionTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ *     limitations under the License.
+ *******************************************************************************/
+
+import static org.junit.Assert.*;
+
+import java.net.MalformedURLException;
+
+import org.junit.Test;
+
+public class ConnectionTest
```
Comment (rpastrana) [inline wsclient/src/test/java/org/hpccsystems/ws/client/utils/ConnectionTest.java:25]: @jchambers-ln not sure how you encountered this issue, if you provide details, we can include junit coverage.

**Comment by rpastrana** [comment]
@jchambers-ln please review.

File: wsclient/src/test/java/org/hpccsystems/ws/client/utils/ConnectionTest.java (hunk 1 above)
Comment (jchambers-ln) [reply wsclient/src/test/java/org/hpccsystems/ws/client/utils/ConnectionTest.java:25]: I saw that even though a https was passed in that the isHttps wasn't set right

**Review by jchambers-ln — APPROVED** [review]
functionality looks good

File: wsclient/src/test/java/org/hpccsystems/ws/client/utils/ConnectionTest.java (hunk 1 above)
Comment (rpastrana) [reply wsclient/src/test/java/org/hpccsystems/ws/client/utils/ConnectionTest.java:25]: ok, in that case the new junit tests should provide enough coverage.

**Pull request merged**